### PR TITLE
Benchmark performance status checks

### LINKED ISSUE #104: Benchmark syrupy performance on test runs in CI
**Is your feature request related to a problem? Please describe.**

It'd be nice to see the impact a diff has on the plugin performance

**Describe the solution you'd like**

- Benchmark code performance
- Compare against head (`master`)
- Commit or stash the generated profile
- Github status/checks on branches to show effect

**Describe alternatives you've considered**

N/A

**Additional context**

- https://github.com/sharkdp/hyperfine
- https://github.com/python/pyperformance
- https://github.com/ionelmc/pytest-benchmark

### PR DESCRIPTION
**Pull request by iamogbz**
## Description

Benchmark syrupy test runs and publish to github as status checks

See https://github.com/tophat/syrupy/commit/eb3cac448f906c1d1198327f4b88455148c08499 for QA results

## Related Issues

- Closes #104 

## Checklist

- [ ] This PR has sufficient test coverage.
- [x] I will merge this pull request with a semantic title.

## Additional Comments

- https://pyperf.readthedocs.io/en/latest/user_guide.html
- https://pygithub.readthedocs.io/en/latest/introduction.html

### CODE REVIEW

**Comment by noahnu** [comment]
Status checks are great but can we post as a comment as well (similar to codecov)?

File: dev_requirements.txt [hunk 1]
```diff
@@ -5,58 +5,63 @@
...
 pycodestyle==2.5.0        # via flake8
 pyflakes==2.1.1           # via flake8
+pygithub==1.47            # via syrupy
 pygments==2.6.1           # via readme-renderer
-pyparsing==2.4.6          # via packaging
-pytest==5.4.1
+pyjwt==1.7.1              # via pygithub
+pyparsing==2.4.7          # via packaging
+pyperf==2.0.0             # via syrupy
+pytest==5.4.1             # via syrupy
```
Comment (noahnu) [inline dev_requirements.txt:48]: That doesn't seem right? It's not a dependency of syrupy

**Comment by iamogbz** [comment]
> Status checks are great but can we post as a comment as well (similar to codecov)?

@noahnu you ask for too much 😞lol will probably expand in a follow up PR, the library has a markdown formatter that works well

File: dev_requirements.txt (hunk 1 above)
Comment (iamogbz) [reply dev_requirements.txt:48]: Not sure why it's doing this yet but will fix before merging

Comment (iamogbz) [reply dev_requirements.txt:48]: Technically it is a dependency of syrupy, it weird but it was introduced in https://github.com/jazzband/pip-tools/releases/tag/4.5.0

**Pull request merged**